renormalizer/mps/svd_qn.py

```diff
@@ -1,10 +1,67 @@
 # -*- coding: utf-8 -*-
 # Author: Jiajun Ren <jiajunren0522@gmail.com>
+import logging
 
 import scipy.linalg
 
 from renormalizer.mps.backend import np
 
+logger = logging.getLogger(__name__)
+
+
+def optimized_svd(a, full_matrices):
+    # optimize performance when ``full_matrices=True`` and the shape of ``a`` is extremely unbalanced
+    # The idea is to construct only a limited number of orthogonal basis rather than all of them
+    # (which are not necessary in most cases)
+    m, n = a.shape
+
+    # whether do the optimization
+    # here 1/3 and 3 are only empirical
+    opt = not (1 / 3 < m / n < 3)
```

Comment:
Why do not add a `opt` flag in `optimized_svd` and `Csvd`? 
if opt is None, opt = not (1 / 3 < m / n < 3). 
In this case, if we need real `full_matrices` in the future, we can just use `opt=False`.

Reply:
Agreed. The `opt` option also serves as documentation for the `Csvd` function.